crates/sats/src/algebraic_type.rs

```diff
@@ -300,6 +327,38 @@ impl AlgebraicType {
             _ => None,
         }
     }
+
+    /// Non-recursively validates that the type is in "nominal normal form".
```

Comment:
This function is in fact recursive, so this should be changed to:
`Validates that the type is in "nominal normal form". Does not follow Refs.`